humanmine.ipynb

```diff
@@ -331,206 +341,390 @@
     "        url = IMAGE_DATA_URL.format(**qs)\n",
     "        url_link = LINK_URL.format(**qs)\n",
-    "        html += '<td><a href=\"'+url_link+'\" target=\"_blank\"><img src=\"'+url+'\"/></a></td>'\n",
+    "        html += '<td><a href=\"'+url_link+'\" target=\"_blank\"><img src=\"'+url+'\"/></a>&nbsp;</td>'\n",
     "    html += \"</tr>\"\n",
-    "html += \"</table>\"\n",
-    "display(HTML(html))"
+    "    html += \"</table>\"\n",
+    "    html_widget.value = html\n",
+    "    \n",
+    "def on_selection_change(change):\n",
+    "    global disease\n",
+    "    if change['name'] == 'value':\n",
+    "        values = get_images(change['new']) \n",
+    "        if values is None:\n",
+    "            return\n",
+    "        disease = change['new']\n",
+    "        min_value = 0\n",
+    "        max_value = increase \n",
+    "        display_images(values, 0, max_value)\n",
+    "                \n",
+    "def get_images(disease):\n",
+    "    for k, v in images:\n",
+    "        if k == disease:\n",
+    "            return v\n",
+    "    return None\n",
+    "    \n",
+    "def on_click_next(b):\n",
+    "    global min_value\n",
+    "    global max_value\n",
+    "    max_value = max_value + increase\n",
+    "    min_value = min_value + increase\n",
+    "    values = get_images(disease)\n",
+    "    button_previous.disabled = False\n",
+    "    if values is None:\n",
+    "        return\n",
+    "    if max_value > len(values):\n",
+    "        button_next.disabled = True\n",
+    "        \n",
+    "    display_images(values, min_value, max_value)\n",
+    "\n",
+    "def on_click_previous(b):\n",
+    "    global min_value\n",
+    "    global max_value\n",
+    "    max_value = max_value - increase\n",
+    "    min_value = min_value - increase\n",
+    "    button_next.disabled = False\n",
+    "    if min_value <= 0:  # reset \n",
+    "        min_value = 0\n",
+    "        max_value = increase\n",
+    "        button_previous.disabled = True\n",
+    "    values = get_images(disease)\n",
+    "    if values is not None:\n",
+    "        display_images(values, min_value, max_value)\n",
+    "            \n",
+    "def dropdown_widget(disease_list,\n",
+    "                    dropdown_widget_name,\n",
+    "                    displaywidget=False):\n",
+    "\n",
+    "    selection = widgets.Dropdown(\n",
+    "        options=disease_list,\n",
+    "        value=disease_list[0],\n",
+    "        description=dropdown_widget_name,\n",
+    "        disabled=False,\n",
+    "    )\n",
+    "    selection.observe(on_selection_change)\n",
+    "    display_images(get_images(selection.value), min_value, max_value)\n",
+    "    return selection\n",
+    "\n",
+    "\n",
+    "\n",
+    "html_widget = widgets.HTML(\"\")\n",
+    "disease_box = dropdown_widget(\n",
+    "    disease_list,\n",
+    "    'Disease: ', True\n",
+    ")\n",
+    "\n",
+    "button_next = widgets.Button(description=\"Next>>\")\n",
```

Comment:
The Next and Previous buttons have a bit erratic behavour. The number of images before and after a sequence of clicks ``Next`` -> ``Previous`` does not necessarily match. Sometimes the clicking of Next or Previous get the widget into a bad state, which can be gotten rid of only by selecting another disease from the dropdown.